CMakeLists.txt

```diff
@@ -64,7 +64,10 @@ if(CMAKE_PROJECT_NAME STREQUAL PROJECT_NAME)
     set(PYTHON_LIBRARIES ${Python_LIBRARIES})
     message("Apple - Using Python:${Python_VERSION_MAJOR} - Libraries:${PYTHON_LIBRARIES} - IncludeDirs: ${PYTHON_INCLUDE_DIR}")
   elseif(UNIX)
-    find_package(PythonLibs 3.9...3.11 REQUIRED)
+    find_package(Python 3.9...<3.12 COMPONENTS Interpreter Development REQUIRED)
+    set(Python_FIND_VIRTUALENV FIRST) # (require cmake >= v3.15 and this is the default) use the Python version configured by pyenv if available
```

Comment:
Can you confirm whether this works with poetry installed virtualenvs too? Also does it break non-pyenv builds?

Reply:
https://cmake.org/cmake/help/latest/module/FindPython.html
`Python_FIND_VIRTUALENV=FIRST` is the default since cmake 3.15, and it would fallback to non-pyenv Python installation